src/sentry/tasks/relay.py

```diff
@@ -46,13 +46,12 @@ def update_config_cache(
     sentry_sdk.set_tag("update_reason", update_reason)
     sentry_sdk.set_tag("generate", generate)
 
-    # Delete key before generating configs such that we never have an outdated
-    # but valid cache.
-    #
-    # If this was running at the end of the task, it would be more effective
-    # against bursts of updates, but introduces a different race where an
-    # outdated cache may be used.
-    projectconfig_debounce_cache.mark_task_done(public_key, project_id, organization_id)
+    # Not running this at the beginning of the task to add tags in case there's
+    # something wrong going on.
+    if not should_update_cache(
+        organization_id=organization_id, project_id=project_id, public_key=public_key
+    ):
+        return
```

Comment:
I remember a concern with returning early here: If we ever enable `acks_late` for this celery task (@untitaker should we?), the task might run a second time when the first attempt crashes. But the second time, the flag is already deleted, so it won't compute the project config.

Reply:
To clarify: I still think this is OK, maybe just add a comment that this would not work with acks_late or retries.

Reply:
I think it's ok for this case to happen, considering the requirements and constraints we have. If the task is marked as done and crashes without computing project configs, relays can always make another request for the project config.

Reply:
That said, I added a comment to clarify it 😄.